src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/runtime/ObjectSynchronizer.java

```diff
@@ -55,7 +55,7 @@ public long identityHashValueFor(Oop obj) {
       // FIXME: can not generate marks in debugging system
       return mark.hash();
     } else if (mark.hasMonitor()) {
-      if (VM.getVM().getCommandLineFlag("LockingMode").getInt() == LockingMode.getLightweight()) {
+      if (VM.getVM().getCommandLineFlag("UseObjectMonitorTable").getBool()) {
```

Comment:
Only with `UseObjectMonitorTable` can we get the hash from the markWord when it has a monitor.